vizro-core/src/vizro/models/_base.py

```diff
@@ -223,9 +224,15 @@ class VizroBaseModel(BaseModel):
         ),
     ]
 
-    @_log_call
-    def model_post_init(self, context: Any) -> None:
-        model_manager[self.id] = self
+    # TODO[MS]: Make more official access to the tree
+    @classmethod
+    def _from_dict_in_build(cls, parent_id: str, field_name: str, data: dict[str, Any]) -> Self:
```

Comment:
I like the idea of something like this to add something to the dashboard tree on the fly but not sure about including the `pre_build` here rather than some alternative like:
* calling manually
* doing another iteration through the dashboard tree (probably not a good idea though)

Two more rough ideas:
* if we put the tree in all models this might become simpler
* can we do it automatically through a pydantic validator when the field is set?